internal/authz/authz.go

```diff
@@ -151,7 +163,17 @@ func (a *Authorizer) Authorize(ctx context.Context, storeID, apiMethod string) e
 		return err
 	}
 
-	return a.individualAuthorize(ctx, claims.ClientID, relation, StoreIDType(storeID).String(), &openfgav1.ContextualTupleKeys{})
+	// Check if there is top-level authorization first, before checking modules
+	err = a.individualAuthorize(ctx, claims.ClientID, relation, StoreIDType(storeID).String(), &openfgav1.ContextualTupleKeys{})
+	if err == nil {
+		return nil
+	}
```

Comment:
Excellent point, i think this should be

```
err1 := individualAuthorize()
if err1 != nil { return err1 }

err2 := moduleAuthorize()
if err2 != nil {return err2 }

return nil
```

IMO `moduleAuthorize` should be the one to check if `len(modules) !=0`

Reply:
@miparnisari the reason I can't check if `individualAuthorize` has an error before `moduleAuthorize` is because `individualAuthorize` checks the authorization at the store level, which we only want to accept immediately if it is authorized. If it returns an error, we then want to check if all the modules are authorized

Reply:
@rhamzeh can you give reasoning about the above logic?

Reply:
Discussed offline, the requirement is a bit counter-intuitive to me: first we want to check the coarse permissions and then the fine-grained ones. 

If that is what we want, please disregard my suggestion :)